### PR TITLE
Expanded `registrations` API. Added test. Updated docs.

### PR DESCRIPTION
**Pull request by arb**
Added more information to the plugin results of `registrations`.

### CODE REVIEW

**Comment by devinivy** [comment]
Any reason the `'0.0.0'` default version is necessary?

**Comment by arb** [comment]
Yes the `0.0.0` default is necessary. If a module doesn't have an `attributes.pkg` object or a `attributes.version`, then the plugin version ends up being `undefined`. Based on the tests and the `default` values in the plugin schema, I think the intent was to set the version to 0.0.0 in any instance where the version is not provided.

**Comment by hueniverse** [comment]
Hmm. The schema has the '0.0.0' default. Is it not working right?

File: lib/plugin.js [hunk 1]
```diff
@@ -264,7 +272,7 @@ internals.Plugin.prototype.register = function (plugins /*, [options], callback
                 Hoek.assert(item.multiple, 'Plugin', item.name, 'already registered');
             }
             else {
-                self.root._registrations[item.name] = { version: item.version };
+                self.root._registrations[item.name] = registrationData;
```
Comment (AdriVanHoudt) [inline lib/plugin.js]: why not just `item` as a whole?

**Comment by arb** [comment]
@hueniverse it works right if your plugin has `register.attributes.pkg = {}`, the the Joi default on "version" pops and defaults it to "0.0.0". In instances where you don't have anything on `pkg`, only the default value of `{}` is set to `pkg`; not on `version`.

**Comment by hueniverse** [comment]
The we need to fix the schema, not the code.

**Comment by arb** [comment]
@hueniverse updated schema so we could remove the `|| '0.0.0'` code change.

File: lib/plugin.js [hunk 2]
```diff
@@ -253,6 +253,14 @@ internals.Plugin.prototype.register = function (plugins /*, [options], callback
         selection.realm.modifiers.route.vhost = item.options.routes.vhost;
         selection.realm.pluginOptions = item.pluginOptions || {};
 
+        var registrationData = {
+            version: item.version,
+            name: item.name,
+            multiple: item.multiple,
```
Comment (hueniverse) [inline lib/plugin.js]: This is redundant given you have attributes.

**Comment by arb** [comment]
This should be good for a second review.

**Pull request merged**